### PR TITLE
Update hostname from required to default_from_api

### PR DESCRIPTION
**Pull request by naitianliu-google**
If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
gkeonprem: updated `hostname` field of ip_block from required to optional from resource `google_gkeonprem_vmware_cluster` 
```

### CODE REVIEW

File: mmv1/products/gkeonprem/VmwareCluster.yaml [hunk 1]
```diff
@@ -232,7 +232,7 @@ properties:
                         description:
                           Hostname of the machine. VM's name will be used if
                           this field is empty.
-                        required: true
+                        default_from_api: true
```
Comment (hao-nan-li) [inline mmv1/products/gkeonprem/VmwareCluster.yaml:235]: May I ask what's the reason of the change?

Comment (naitianliu-google) [reply mmv1/products/gkeonprem/VmwareCluster.yaml:235]: Thanks for asking. This is a fix on the initial resource schema. The hostname should be optional and defaulted on server side. It is only required to provide ip address for static ip config. @hao-nan-li

Comment (hao-nan-li) [reply mmv1/products/gkeonprem/VmwareCluster.yaml:235]: Thanks for the explanation!

**Review by hao-nan-li — APPROVED** [review]
LGTM
